hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
hmm, I am probably not seeing sth, but from your comment, even if the return value is UNKNOWN (instead of the codebase one), that would mean that we never get to exercise the EMPTY_PWD path, since UNKNOWN=0 which is always less then 3.9, and since this operation happens before cluster join, then the else path will never happen. Missing something ?